### ISSUE TITLE
new behavior regarding `Series.plot.bar` for `logy=True` in 0.21

### ISSUE THREAD

**Issue by pcluo** [pandas-dev/pandas#18394]
#### Code Sample
```python
Series([10**x for x in range(5)]).plot.bar()
Series([10**x for x in range(5)]).plot.bar(logy=True)
```
#### Problem description

The behavior changed from last version, 0.20. In last version, `Series.plot.bar` uses one color for bars. In the new version, it uses different colors for bars even for a single series. When using `logy` argument, the bars cannot be seen. It seems the fill color is same as the background color.

![image](https://user-images.githubusercontent.com/1349042/33044244-31e3451e-ce16-11e7-9c2e-ca86ac0664ec.png)
![image](https://user-images.githubusercontent.com/1349042/33044250-363cfc68-ce16-11e7-9dec-1b115e399369.png)

#### Output of ``pd.show_versions()``

<details>

INSTALLED VERSIONS
------------------
commit: None
python: 3.6.3.final.0
python-bits: 64
OS: Windows
OS-release: 10
machine: AMD64
processor: Intel64 Family 6 Model 94 Stepping 3, GenuineIntel
byteorder: little
LC_ALL: None
LANG: None
LOCALE: None.None

pandas: 0.21.0
pytest: 3.2.1
pip: 9.0.1
setuptools: 36.5.0.post20170921
Cython: 0.26.1
numpy: 1.13.3
scipy: 1.0.0
pyarrow: 0.7.0
xarray: None
IPython: 6.1.0
sphinx: 1.6.3
patsy: 0.4.1
dateutil: 2.6.1
pytz: 2017.2
blosc: None
bottleneck: 1.2.1
tables: 3.4.2
numexpr: 2.6.2
feather: 0.4.0
matplotlib: 2.1.0
openpyxl: 2.4.8
xlrd: 1.1.0
xlwt: 1.3.0
xlsxwriter: 1.0.2
lxml: 4.1.0
bs4: 4.6.0
html5lib: 0.999999999
sqlalchemy: 1.1.13
pymysql: 0.7.11.None
psycopg2: None
jinja2: 2.9.6
s3fs: None
fastparquet: None
pandas_gbq: None
pandas_datareader: 0.5.0]

</details>

**Comment by PMeira** [comment]
Bumped into the same issue. You can pass a `bottom` argument to fix it, e.g.

`Series([10**x for x in range(5)]).plot.bar(logy=True, bottom=1)`

**Comment by Deborah-Digges** [comment]
@PMeira Why was the behavior of Series.plot.bar changed to plot bars with different color. Visually these colors add nothing and add to clutter in the chart. Different colors should only be used when they correspond to differences of meaning in the data.

Why is the default behavior to provide an unnecessarily visually overwhelming graph? It took me some time to realize why my bars suddenly started acting strangely. Now, I pass color='cornflowerblue' to get all my bars the same pleasant hue.

Reference for visual appeal and the use of colors: http://www.perceptualedge.com/articles/visual_business_intelligence/rules_for_using_color.pdf

**Comment by PMeira** [comment]
@Deborah-Digges I completely agree, I don't quite understand the change either. Maybe a dev can enlighten us.

**Comment by Deborah-Digges** [comment]
Thanks @PMeira. apologies, I assumed you were a dev!

**Comment by TomAugspurger** [comment]
Sorry about missing this when the issue was opened.

Is anyone interested in debugging this to see where things break? We're doing a release in 2-3 weeks, would be nice to get this fixed.

**Comment by TomAugspurger** [comment]
Actually, the `log` issue seems to be fixed on master

![gh](https://user-images.githubusercontent.com/1312546/38207229-acbbb6b2-3672-11e8-84e3-a45596798161.png)


Tests confirming that this is fixed would be welcome.

**Comment by Deborah-Digges** [comment]
@TomAugspurger The issue is really different than the one just closed as a duplicate. The reason I raised it as a separate issue is because it's more a question of visual encoding than broken functionality. There's probably a reason the bars were made differently colored post version 0.20 and that's something I want to understand, which seems unrelated to the broken functionality that resulted because of this change(described above)

**Comment by pcluo** [comment]
I should be more clear in my initial post. Visual encoding is a design choice while the `log` issue was a bug. However, I agree with @Deborah-Digges  and do prefer a single color when plotting one variable. Right now, I use `Series.plot(color='C0')` to get default behavior in previous versions. 

For the `log` issue, I'm not entirely sure how to `assert` an output that is a graph. Otherwise, I am happy to produce a test later this week @TomAugspurger.

**Comment by TomAugspurger** [comment]
For the test, I'd imagine that in the broken version, `ax.patches[-1].get_height()` would be some small number. Ideally you'd write a test that checks it's the correct height, ensure that the test fails on the broken version, test that it works on the new version, then make a PR with the test.

That said, I can't reproduce this even with pandas 0.21.0. We'll want to make sure the issue is reproducible before writing a test for it.

**Comment by pcluo** [comment]
@TomAugspurger the `logy` issue seems to be related to matplotlib. I can reproduce it with pandas 0.21.0 but matplotlib 2.1.0, which is the installed version in my first post, but not with the newer matplotlib 2.2.2.

**Comment by jtweeder** [comment]
@TomAugspurger , This may be old and left open, but you had previously asked for more tests that this is fixed.  It appears that @pcluo is correct about the matplotlib issue.

I tested with matplotlib 2.1.0 and pandas 0.22.0 and reproduced a similar issue
![oldversuion](https://user-images.githubusercontent.com/31365056/45718897-45fafb00-bb64-11e8-8ab5-a532a317ddf5.PNG)

Then I tested matplotlib 2.2.2 and pandas 0.22.0 and no issue.
![newversion](https://user-images.githubusercontent.com/31365056/45718990-907c7780-bb64-11e8-8ae6-15aeda54bdaf.PNG)


I also had matplotlib 2.2.2 and pandas 0.20.3 and had no issue, so I think matplotlib may be the culprit.
![third](https://user-images.githubusercontent.com/31365056/45719074-cd486e80-bb64-11e8-8f80-88d5adb64676.PNG)

Hope that provides some useful information!
